test/scope_token_to_list_of_private_public_repos_test.go

```diff
@@ -0,0 +1,160 @@
+//go:build e2e
+// +build e2e
+
+package test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	pacv1alpha1 "github.com/openshift-pipelines/pipelines-as-code/pkg/apis/pipelinesascode/v1alpha1"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/params"
+	tgithub "github.com/openshift-pipelines/pipelines-as-code/test/pkg/github"
+	"github.com/openshift-pipelines/pipelines-as-code/test/pkg/options"
+	"github.com/openshift-pipelines/pipelines-as-code/test/pkg/payload"
+	trepo "github.com/openshift-pipelines/pipelines-as-code/test/pkg/repository"
+	twait "github.com/openshift-pipelines/pipelines-as-code/test/pkg/wait"
+	"github.com/tektoncd/pipeline/pkg/names"
+	"gotest.tools/v3/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestGithubPullRequestScopeTokenToListOfRepos(t *testing.T) {
+	if os.Getenv("NIGHTLY_E2E_TEST") != "true" {
+		t.Skip("Skipping test since only enabled for nightly")
+	}
+	targetNS := names.SimpleNameGenerator.RestrictLengthWithRandomSuffix("pac-e2e-ns")
+	ctx := context.Background()
+	runcnx, opts, ghcnx, err := tgithub.Setup(ctx, false)
+	assert.NilError(t, err)
+
+	// update config map with secret-github-app-token-scoped to false
+	updateAndRevertCM(ctx, t, runcnx, "false")
+
+	var remoteTaskURL, remoteTaskName string
+	if os.Getenv("TEST_GITHUB_PRIVATE_TASK_URL") != "" {
+		remoteTaskURL = os.Getenv("TEST_GITHUB_PRIVATE_TASK_URL")
+	}
+	if os.Getenv("TEST_GITHUB_PRIVATE_TASK_NAME") != "" {
+		remoteTaskName = os.Getenv("TEST_GITHUB_PRIVATE_TASK_NAME")
```

Comment:
whhat if the value is not set ?

Reply:
As the test case focus is to test task from private repo so providing env is mandatory
If in case env not provided test case will fail
```
$ go test -tags=e2e -v -run ^TestGithubPullRequestScopeTokenToListOfRepos$
=== RUN   TestGithubPullRequestScopeTokenToListOfRepos
    github_scope_token_to_list_of_private_public_repos_test.go:41: Env TEST_GITHUB_PRIVATE_TASK_URL not provided
--- FAIL: TestGithubPullRequestScopeTokenToListOfRepos (6.91s
```

Reply:
sounds good